github/client.go

```diff
@@ -177,6 +187,75 @@ func (c *Client) Token(ctx context.Context, opts *tokenOptions) (*logical.Respon
 	return tokenRes, nil
 }
 
+type Installation struct {
+	ID      int     `json:"id"`
+	Account Account `json:"account"`
+}
+
+type Account struct {
+	Login string `json:"login"`
+}
+
+func (c *Client) getInstallationID(config *Config) (int, error) {
+	expires := jwt.NewNumericDate(time.Now().Add(time.Second * time.Duration(120)))
+	issuedAt := jwt.NewNumericDate(time.Now().Add(time.Second * -10))
+	claims := &jwt.RegisteredClaims{
+		ExpiresAt: expires,
+		IssuedAt:  issuedAt,
+		Issuer:    strconv.Itoa(config.AppID),
+	}
+	signKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(config.PrvKey))
+	if err != nil {
+		return 0, err
+	}
+
+	instURL, err := url.ParseRequestURI(fmt.Sprintf(
+		"%s/app/installations",
+		strings.TrimSuffix(config.BaseURL, "/"),
+	))
+	if err != nil {
+		return 0, err
+	}
+	signedToken, err := jwt.NewWithClaims(jwt.SigningMethodRS256, claims).SignedString(signKey)
+	if err != nil {
+		return 0, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, instURL.String(), nil)
+	if err != nil {
+		return 0, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", signedToken))
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+
+	// Perform the request, re-using the shared transport.
+	res, err := c.transport.RoundTrip(req)
+	if err != nil {
+		return 0, fmt.Errorf("%w: RoundTrip error: %v", errUnableToCreateAccessToken, err)
+	}
+
+	defer res.Body.Close()
+
+	var instResult []Installation
+	if err := json.NewDecoder(res.Body).Decode(&instResult); err != nil {
+		return 0, err
+	}
+
+	var instID int
+	for _, v := range instResult {
+		if v.Account.Login == config.OrgName {
+			instID = v.ID
```

Comment:
You can return from here and make replace code after the cycle with just
```go
return 0, fmt.Errorf("installation ID for the organization wasn't found")
```

Maybe also reword error message? Mentioning app not installed in the organization.